etc/kaleidoscope-builder.conf

```diff
@@ -12,10 +12,26 @@ LIBRARY="${LIBRARY:-${SKETCH}}"
 
 BOARD="${BOARD:-model01}"
 MCU="${MCU:-atmega32u4}"
-if [ "${BOARD}" = "virtual" ]; then
+
+if [ -z "${ARCH}" ]; then
+   ARCH=$(echo "${FQBN}" | sed -n -e 's/^[^:]\+:\([^:]\+\).*/\1/p')
+fi
+
+if [ "${ARCH}" = "x86" ]; then
     FQBN="${FQBN:-keyboardio:x86:${BOARD}}"
+    
+    # Set the compiler path for virtual builds (x86)
+    #
+    if [ -z "${COMPILER_PATH}" ]; then
```

Comment:
For virtual builds we automatically set compiler information, unless predefined.